src/transformers/models/marian/modeling_marian.py

```diff
@@ -823,7 +829,8 @@ def __init__(self, config: MarianConfig, embed_tokens: Optional[nn.Embedding] =
         if embed_tokens is not None:
             self.embed_tokens = embed_tokens
         else:
-            self.embed_tokens = nn.Embedding(config.vocab_size, config.d_model, self.padding_idx)
+            # TODO: can we assume if embeddings are not shared the vocabs are also not shared ?
```

Comment:
I think so yes. Also since we let `config.decoder_vocab_size` default to `config.vocab_size` I don't think that this is a problem no?